File: example3/src/FibonacciSeries.py

def print_fibonacci_series(n):
    n1, n2 = 0, 1
    print(n1, n2, end=' ')
    
    for _ in range(2, n):
        n1, n2 = n2, n1 + n2
        print(n2, end=' ')
    
    print()

if __name__ == "__main__":
    count = 10
    print("Fibonacci Series of", count, "numbers:")
    print_fibonacci_series(count)

File: example3/src/FibonacciSeries.hs

module Main where

fibonacci :: Int -> [Int]
fibonacci n = take n fibs
  where fibs = 0 : 1 : zipWith (+) fibs (tail fibs)

main :: IO ()
main = do
  let count = 10
  putStrLn $ "Fibonacci Series of " ++ show count ++ " numbers: " ++ show (fibonacci count)
